File: easy/2974.py
"""2974. Minimum Number Game"""

from typing import List


class Solution:
    def numberGame(self, nums: List[int]) -> List[int]:
        nums.sort()

        out: list[int] = []

        for idx in range(0, len(nums), 2):
            out.append(nums[idx + 1])
            out.append(nums[idx])

        return out

File: easy/2974.rs
// 2974. Minimum Number Game

impl Solution {
    pub fn number_game(nums: Vec<i32>) -> Vec<i32> {
        let mut sorted_nums = nums.clone();
        sorted_nums.sort_unstable();

        let mut out = Vec::with_capacity(sorted_nums.len());

        for idx in (0..sorted_nums.len()).step_by(2) {
            out.push(sorted_nums[idx + 1]);
            out.push(sorted_nums[idx]);
        }

        out
    }
}
